### PR TITLE
[rpclib] Update to 2.3.0

### PR DESCRIPTION
**Pull request by chausner**
**Describe the pull request**

- #### What does your PR fix?  
  Updates rpclib to 2.3.0

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  As before, Yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run ./vcpkg x-add-version --all and committed the result?  
  Yes

### CODE REVIEW

File: ports/rpclib/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "rpclib",
-  "version-string": "2.2.1",
-  "port-version": 3,
+  "version-semver": "2.3.0",
```
Comment (chausner) [inline ports/rpclib/vcpkg.json:3]: Switched to semver as according to past releases (https://github.com/rpclib/rpclib/releases), the library appears to follow semantic versioning.

File: ports/rpclib/portfile.cmake [hunk 2]
```diff
@@ -3,23 +3,24 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
...
     PREFER_NINJA
 )
 
 vcpkg_install_cmake()
 
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
 
-vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/rpclib)
+vcpkg_fixup_cmake_targets(CONFIG_PATH "lib/cmake/rpclib")
```
Comment (NancyLi1013) [inline ports/rpclib/portfile.cmake:21]:
```suggestion
vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/rpclib)
```
Quotes are not needed for a relative path.

File: ports/rpclib/portfile.cmake [hunk 3]
```diff
@@ -3,23 +3,24 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
...
 
 vcpkg_copy_pdbs()
 
-file(INSTALL ${SOURCE_PATH}/LICENSE.md DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
+file(INSTALL "${SOURCE_PATH}/LICENSE.md" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
+
```
Comment (NancyLi1013) [inline ports/rpclib/portfile.cmake:26]:
```suggestion
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/rpclib/portfile.cmake (hunk 2 above)
Comment (NancyLi1013) [reply ports/rpclib/portfile.cmake:21]: Through internal discussion, adding the quotes or not are both okay, so it's unnecessary to update this.

File: versions/r-/rpclib.json [hunk 4]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "a6168697e9d189226609d6879db5ac25a1b4a4c5",
```
Comment (NancyLi1013) [inline versions/r-/rpclib.json:4]:
```suggestion
      "git-tree": "8479e995436633dafc97a4dd951beff832bc3367",
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Review by NancyLi1013 — APPROVED** [review]

**Pull request merged**